### PR TITLE
feat: store edv vaultID in issuer profiles

### LINKED ISSUE #491: Once EDV no longer allows the client to pick vault name, incorporate that change here
After https://github.com/trustbloc/edv/issues/112 is done, update edge-service and store the vault names somewhere.

### PR DESCRIPTION
**Pull request by grace0925**
closes #491 
Signed-off-by: Grace Liu <grace.liu@securekey.com>

### CODE REVIEW

**Review by fqutishat — APPROVED** [review]

File: pkg/restapi/issuer/operation/operations.go [hunk 1]
```diff
@@ -505,7 +499,14 @@ func (o *Operation) retrieveCredentialHandler(rw http.ResponseWriter, req *http.
 		}
 	}
 
-	o.retrieveCredential(rw, profile, docURLs)
+	o.retrieveCredential(rw, profileName, profile.EDVVaultID, docURLs)
+}
+
+func (o *Operation) getVaultIDFromVaultLocationURL(vaultLocationURL string) string {
```
Comment (sudeshrshetty) [inline pkg/restapi/issuer/operation/operations.go:505]: since this function is not accessing any elements from `Operation` type, you can move it out.

Comment (sudeshrshetty) [reply pkg/restapi/issuer/operation/operations.go:505]: can be shortened `return vaultLocationURLSplitUp[len(vaultLocationURLSplitUp)-1]`

File: pkg/restapi/issuer/operation/operations.go [hunk 2]
```diff
@@ -519,12 +520,29 @@ func (o *Operation) createIssuerProfile(pr *ProfileRequest) (*vcprofile.DataProf
 
 	created := time.Now().UTC()
 
+	edvVaultID, err := o.createIssuerProfileVault(pr.Name)
+	if err != nil {
+		return nil, err
```
Comment (DRK3) [inline pkg/restapi/issuer/operation/operations.go:525]: I suggest wrapping this error - it will make debugging any issues later easier

File: pkg/restapi/issuer/operation/operations.go [hunk 3]
```diff
@@ -519,12 +520,29 @@ func (o *Operation) createIssuerProfile(pr *ProfileRequest) (*vcprofile.DataProf
...
+		SignatureType: pr.SignatureType, SignatureRepresentation: pr.SignatureRepresentation, EDVVaultID: edvVaultID,
+		Creator: publicKeyID, DisableVCStatus: pr.DisableVCStatus, OverwriteIssuer: pr.OverwriteIssuer,
 	}, nil
 }
 
+// createIssuerProfileVault creates the vault associated with the profile
+func (o *Operation) createIssuerProfileVault(profileName string) (string, error) {
+	vaultLocationURL, err := o.edvClient.CreateDataVault(&models.DataVaultConfiguration{ReferenceID: profileName})
+	if err != nil {
+		return "", err
```
Comment (DRK3) [inline pkg/restapi/issuer/operation/operations.go:538]: I suggest wrapping this error - it will make debugging any issues later easier

File: pkg/restapi/issuer/operation/operations.go [hunk 4]
```diff
@@ -652,7 +670,6 @@ func (o *Operation) composeAndIssueCredentialHandler(rw http.ResponseWriter, req
 
 		return
 	}
-
```
Comment (DRK3) [inline pkg/restapi/issuer/operation/operations.go:655]: You should remove these unrelated formatting changes (removal of line breaks) in order to keep your PR focused on the one thing it says it does

File: pkg/restapi/issuer/operation/operations_test.go [hunk 5]
```diff
@@ -496,6 +496,31 @@ func testCreateProfileHandler(t *testing.T) {
 
 		require.Equal(t, invalidRequestErrMsg+": EOF", errResp.Message)
 	})
+	t.Run("create profile error unable to create vault in edv", func(t *testing.T) {
```
Comment (DRK3) [inline pkg/restapi/issuer/operation/operations_test.go:499]: Add a `-` or `:` between the "create profile error" and the "unable" in order to make it easier to read

Comment (DRK3) [inline pkg/restapi/issuer/operation/operations_test.go:499]: `edv` should be capitalized

File: test/bdd/fixtures/edv-rest/.env [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
-EDV_REST_IMAGE_TAG=0.1.4
+EDV_REST_IMAGE=docker.pkg.github.com/trustbloc-cicd/snapshot/edv-rest
+EDV_REST_IMAGE_TAG=0.1.5-snapshot-f2cf57a
 
 EDV_HOST=0.0.0.0
 EDV_PORT=8071
 EDV_DATABASE_TYPE=couchdb
 EDV_DATABASE_URL=couchdb.example.com:5984
-EDV_DATABASE_PREFIX=edv
+EDV_DATABASE_PREFIX=edv
```
Comment (DRK3) [inline test/bdd/fixtures/edv-rest/.env:14]: Looks like you accidentally removed the newline at the end of the file here.

**Review by DRK3 — CHANGES_REQUESTED** [review]
Some minor issues to fix.

File: pkg/restapi/issuer/operation/operations.go (hunk 3 above)
Comment (DRK3) [reply pkg/restapi/issuer/operation/operations.go:538]: (note that these "wrap error" comments apply to other places as well where you have unwrapped errors

File: pkg/restapi/issuer/operation/operations.go (hunk 1 above)
Comment (grace0925) [reply pkg/restapi/issuer/operation/operations.go:505]: fixed!

File: pkg/restapi/issuer/operation/operations.go (hunk 2 above)
Comment (grace0925) [reply pkg/restapi/issuer/operation/operations.go]: fixed!

File: pkg/restapi/issuer/operation/operations.go (hunk 3 above)
Comment (grace0925) [reply pkg/restapi/issuer/operation/operations.go]: fixed!

File: pkg/restapi/issuer/operation/operations_test.go (hunk 5 above)
Comment (grace0925) [reply pkg/restapi/issuer/operation/operations_test.go]: fixed!

Comment (grace0925) [reply pkg/restapi/issuer/operation/operations_test.go]: fixed!

File: test/bdd/fixtures/edv-rest/.env (hunk 6 above)
Comment (grace0925) [reply test/bdd/fixtures/edv-rest/.env]: fixed!

File: pkg/restapi/issuer/operation/operations.go (hunk 4 above)
Comment (grace0925) [reply pkg/restapi/issuer/operation/operations.go]: fixed!

**Review by sudeshrshetty — APPROVED** [review]
